src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -702,6 +725,111 @@ private async Task TryNotifyClientErrorAsync(IClientProxy client, string error,
+                    throw new InvalidOperationException("UpdateRootComponents is not supported when components have" +
+                        " been provided during circuit start up.");
+                }
+                if (_isFirstUpdate)
+                {
+                    _isFirstUpdate = false;
+                    if (store != null)
+                    {
+                        shouldClearStore = true;
+                        // We only do this if we have no root components. Otherwise, the state would have been
+                        // provided during the start up process
+                        var appLifetime = _scope.ServiceProvider.GetRequiredService<ComponentStatePersistenceManager>();
+                        await appLifetime.RestoreStateAsync(store);
+                    }
+
+                    // Retrieve the circuit handlers at this point.
+                    _circuitHandlers = [.. _scope.ServiceProvider.GetServices<CircuitHandler>().OrderBy(h => h.Order)];
+                    await OnCircuitOpenedAsync(cancellation);
+                    await OnConnectionUpAsync(cancellation);
+
+                    for (var i = 0; i < operations.Length; i++)
+                    {
+                        var operation = operations[i];
+                        if (operation.Item1.Type != RootComponentOperationType.Add)
+                        {
+                            throw new InvalidOperationException($"The first set of update operations must always be of type {nameof(RootComponentOperationType.Add)}");
+                        }
+                    }
+
+                    pendingTasks = new Task[operations.Length];
+                }
+
+                for (var i = 0; i < operations.Length;i++)
+                {
+                    var (operation, descriptor) = operations[i];
+                    switch (operation.Type)
+                    {
+                        case RootComponentOperationType.Add:
+                            var task = Renderer.AddComponentAsync(descriptor.ComponentType, descriptor.Parameters, operation.SelectorId.Value.ToString(CultureInfo.InvariantCulture));
+                            if (pendingTasks != null)
+                            {
+                                pendingTasks[i] = task;
+                            }
+                            break;
+                        case RootComponentOperationType.Update:
+                            var componentType = Renderer.GetExistingComponentType(operation.ComponentId.Value);
+                            if (descriptor.ComponentType != componentType)
+                            {
+                                Log.InvalidComponentTypeForUpdate(_logger, message: "Component type mismatch.");
+                            }
+
+                            _ = Renderer.UpdateRootComponentAsync(operation.ComponentId.Value, descriptor.Parameters);
```

Comment:
Why is it OK to discard the task here?

Reply:
Any exception that happens will be dispatched through the UnhandledExceptionHandler in the remote renderer

Reply:
OK, could you add a comment to that effect?